greenpithumb/db_store.py

```diff
@@ -41,17 +41,20 @@ def latest_soil_moisture(self):
             return results[0][0]
 
     def retrieve_soil_moisture(self):
-        """Returns timestamp and soil moisture readings.
+        """Retrieves timestamp and soil moisture readings.
 
-        Return values are a list of namedtuples with 'timestamp' and
-        'soil_moisture' fields.
+        Returns:
+            A list of objects with 'timestamp' and 'soil_moisture' fields.
         """
-        self._cursor.execute("SELECT * FROM soil_moisture")
+        self._cursor.execute('SELECT * FROM soil_moisture')
         SoilMoistureRecord = collections.namedtuple(
             'SoilMoistureRecord', ['timestamp', 'soil_moisture'])
         soil_moisture_data = map(SoilMoistureRecord._make,
                                  self._cursor.fetchall())
-        return soil_moisture_data
+        if soil_moisture_data == []:
+            return None
+        else:
+            return soil_moisture_data
```

Comment:
(optional) This is just a style thing but I like avoiding nesting when I can, so if I ever have a thing of like 

```python
if whatever:
  return x
else:
  return y
```

I'd rather just do

```python:
if whatever:
  return x
return y
```